src/services/activation/model.js

```diff
@@ -0,0 +1,59 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+const { Configuration } = require('@schul-cloud/commons');
+const crypto = require('crypto');
+const { enableAuditLog } = require('../../utils/database');
+
+const {
+	KEYWORDS,
+	STATE,
+	createQuarantinedObject,
+	getQuarantinedObject,
+} = require('./utils');
+
+/**
+ * WARNING: Document will be removed after X (ACTIVATION_LINK_PERIOD_OF_VALIDITY_SECONDS) seconds
+ */
+const activationSchema = new Schema({
+	activationCode: { type: String, required: true },
+	userId: { type: Schema.Types.ObjectId, required: true, ref: 'user' },
+	keyword: { type: String, required: true, enum: Object.values(KEYWORDS) },
+	quarantinedObject: { type: Object, required: true },
+	mailSent: { type: [Date] },
+	state: { type: String, default: STATE.NOT_STARTED, enum: Object.values(STATE) },
+}, { timestamps: true });
+activationSchema.index({ activationCode: 1 }, { unique: true });
+activationSchema.index({ userId: 1, keyword: 1 }, { unique: true });
+activationSchema.index(
+	{ createdAt: 1 },
+	{ expireAfterSeconds: Configuration.get('ACTIVATION_LINK_PERIOD_OF_VALIDITY_SECONDS') },
```

Comment:
What happens if we change the env value? The index will be created only by migration in production.

Reply:
That's right, the index would not be adjusted in the DB :/ Guess I didn't consider that. 
If you shorten the time, the activation code is valid for a shorter time, because it also works with the variable. But if you extend it, it will be removed from the db, even though it would still be valid.

Reply:
maybe the entry should be in the database for up to 30 days, regardless of the variable and the validity period of the activation link?

Reply:
or we could use [syncIndexes](https://mongoosejs.com/docs/api/model.html#model_Model.syncIndexes)

Reply:
I would expect changes to only apply to new entries. An entry that was created at a time where entries lasted two hours, should last two hours, because thats what the user has been told.

Reply:
have you considered storing these in redis if available?

Reply:
No, not really

Reply:
For now, I'm gonna stick with mongo.